scripts/partition_manager.py

```diff
@@ -222,6 +222,30 @@ def convert_str_to_list(with_str):
             with_str[k].append(v)
 
 
+def resolve_ambiguous_requirements(reqs, unsolved):
+    """
+    Find partitions where the requirements are identical, and therefore
+    ambiguous. For all partitions with identical requirements, introduce
+    requirements so that the partitions have unique placements(sorted by name).
+    """
+
+    buckets = dict()
+    for p in unsolved:
```

Comment:
Can't this also just be called partitions?
Since you are using `for partitions in buckets.values()`

Reply:
Its a single partition, so I changed it to `partition` :)